backend/datasource/query_type_mux.go

```diff
@@ -0,0 +1,75 @@
+package datasource
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+// The QueryTypeHandlerFunc type is an adapter to allow the use of
+// ordinary functions as backend.QueryDataHandler. If f is a function
+// with the appropriate signature, QueryTypeHandlerFunc(f) is a
+// Handler that calls f.
+type QueryTypeHandlerFunc func(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error)
+
+// QueryData calls f(ctx, req).
+func (fn QueryTypeHandlerFunc) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	return fn(ctx, req)
+}
+
+// QueryTypeMux is a query type multiplexer.
+type QueryTypeMux struct {
+	m map[string]backend.QueryDataHandler
+}
+
+// NewQueryTypeMux allocates and returns a new QueryTypeMux.
+func NewQueryTypeMux() *QueryTypeMux {
+	return new(QueryTypeMux)
+}
+
+// Handle registers the handler for the given query type.
+// If a handler already exists for query type, Handle panics.
+func (mux *QueryTypeMux) Handle(queryType string, handler backend.QueryDataHandler) {
+	if queryType == "" {
+		panic("datasource: invalid query type")
+	}
+	if handler == nil {
+		panic("datasource: nil handler")
+	}
+	if _, exist := mux.m[queryType]; exist {
+		panic("datasource: multiple registrations for " + queryType)
+	}
+
+	if mux.m == nil {
```

Comment:
Should this be in NewQueryTypeMux?

Reply:
You can do &QueryTypeMux{} as well because struct is public. Looked at http.ServeMyx and they did the same thing.